<launch>
   <arg name="bbox_size_x" default="200"/>
   <arg name="bbox_size_y" default="200"/>
   <arg name="bbox_pos_x" default="200"/>
   <arg name="bbox_pos_y" default="200"/>
   <arg name="max_nb_iter_per_level" default="10"/>
   <arg name="max_nb_pyr_level" default="3"/>
   <arg name="sampling_rate" default="1.0"/>
   <arg name="camera_serial" default="0" />
   <arg name="calibrated" default="0" />
   <arg name="use_ptgrey" default="false" />
   <arg name="reference_image_path" default="/home/mestrado/Desktop/ref_img.png"/>


   <group if="$(arg use_ptgrey)" ns="usb_cam">
      <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />

      <node pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
         <param name="frame_id" value="camera_link" />
      <!--param name="serial" value="$(arg camera_serial)" /-->
      <!--param name="format7_color_coding" value="raw16" /-->
      <!-- <param name="video_mode" value="1280x960_mono8" /> -->
         <param name="frame_rate" value="60" />
         <param name="format7_roi_width" value="1280" />
         <param name="format7_roi_height" value="960" />
         <param name="format7_x_offset" value="30" />
         <param name="format7_y_offset" value="30" />

      
      <!-- Use the camera_calibration package to create this file -->
      <!-- param name="camera_info_url" if="$(arg calibrated)"
           value="file://$(env HOME)/.ros/camera_info/$(arg camera_serial).yaml" / -->
      </node>

      <node pkg="nodelet" type="nodelet" name="image_proc_debayer"
         args="load image_proc/debayer camera_nodelet_manager">
      </node>
   </group>

   <group unless="$(arg use_ptgrey)" >
      <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
         <param name="video_device" value="/dev/video0" />
         <param name="image_width" value="640" />
         <param name="image_height" value="480" />
         <param name="pixel_format" value="yuyv" />
         <param name="camera_frame_id" value="usb_cam" />
         <param name="io_method" value="mmap"/>
      </node>
   </group>
   
   <node name="tracker_node" pkg="visual_tracking" type="esm_tracker_node" output="screen">
      <param name="image_topic" value="usb_cam/image_raw"/>
      <param name="bbox_size_x" value="$(arg bbox_size_x)"/>
      <param name="bbox_size_y" value="$(arg bbox_size_y)"/>
      <param name="bbox_pos_x" value="$(arg bbox_pos_x)"/>
      <param name="bbox_pos_y" value="$(arg bbox_pos_y)"/>
      <param name="max_nb_iter_per_level" value="$(arg max_nb_iter_per_level)"/>
      <param name="max_nb_pyr_level" value="$(arg max_nb_pyr_level)"/>
      <param name="sampling_rate" value="$(arg sampling_rate)"/>
      <param name="reference_image_path" value="$(arg reference_image_path)"/>
   </node>

   <node name="tracker_output" pkg="image_view" type="image_view" respawn="false" output="screen">
     <remap from="image" to="annotated_image"/>
     <param name="autosize" value="false" />
   </node>

   <node name="stabilized_output" pkg="image_view" type="image_view" respawn="false" output="screen">
     <remap from="image" to="stabilized_image"/>
     <param name="autosize" value="false" />
   </node>
</launch>